modules/recast/navigation_mesh_generator.cpp

```diff
@@ -294,11 +296,22 @@ void EditorNavigationMeshGenerator::_convert_detail_mesh_to_native_navigation_me
 	}
 }
 
-void EditorNavigationMeshGenerator::_build_recast_navigation_mesh(Ref<NavigationMesh> p_nav_mesh, EditorProgress *ep,
+#ifdef TOOLS_ENABLED
+void EditorNavigationMeshGenerator::_build_recast_navigation_mesh(Ref<NavigationMesh> p_nav_mesh, 
 		rcHeightfield *hf, rcCompactHeightfield *chf, rcContourSet *cset, rcPolyMesh *poly_mesh, rcPolyMeshDetail *detail_mesh,
-		Vector<float> &vertices, Vector<int> &indices) {
-	rcContext ctx;
-	ep->step(TTR("Setting up Configuration..."), 1);
+		Vector<float> &vertices, Vector<int> &indices , EditorProgress *ep = NULL) {
+#else
+void EditorNavigationMeshGenerator::_build_recast_navigation_mesh(Ref<NavigationMesh> p_nav_mesh,
+			rcHeightfield * hf, rcCompactHeightfield * chf, rcContourSet * cset, rcPolyMesh * poly_mesh, rcPolyMeshDetail * detail_mesh,
+			Vector<float> & vertices, Vector<int> & indices) {
+#endif
+
+rcContext ctx;
+
+#ifdef TOOLS_ENABLED
+	if (ep)
```

Comment:
To follow the code style guidelines, please replace all instances of `if(ep)` with:

```cpp
if (ep) {
```

Braces should now always be used for single-line `if` conditions.